state/upgrades.go

```diff
@@ -1212,52 +1212,53 @@ func migrateModelLeasesToGlobalTime(st *State) error {
 	coll, closer := st.db().GetCollection(leasesC)
 	defer closer()
 
-	var doc struct {
-		DocID     string `bson:"_id"`
-		Type      string `bson:"type"`
-		Namespace string `bson:"namespace"`
-		Name      string `bson:"name"`
-		Holder    string `bson:"holder"`
-		Expiry    int64  `bson:"expiry"`
-		Writer    string `bson:"writer"`
-	}
-
 	// Find all old lease/clock-skew documents, remove them
 	// and create replacement lease docs in the new format.
 	//
 	// Replacement leases are created with a duration of a
 	// minute, relative to the global time epoch.
-	var ops []txn.Op
-	iter := coll.Find(bson.D{{"type", bson.D{{"$exists", true}}}}).Iter()
-	for iter.Next(&doc) {
-		ops = append(ops, txn.Op{
-			C:      leasesC,
-			Id:     st.localID(doc.DocID),
-			Assert: txn.DocExists,
-			Remove: true,
-		})
-		if doc.Type != "lease" {
-			continue
+	err := st.db().Run(func(int) ([]txn.Op, error) {
+		var doc struct {
+			DocID     string `bson:"_id"`
```

Comment:
why not leave this as an old type instead of being an embedded struct. type oldLeaseDoc or something like that?

Reply:
Why have a named type when you don't need one? I just cribbed from another upgrade. Doesn't really matter either way.